inc/composer/class-docker-compose-generator.php

```diff
@@ -0,0 +1,513 @@
+				],
+				'environment' => [
+					'AWS_REGION' => 'us-east-1',
+					'AWS_S3_BUCKET' => "s3-{$this->project_name}",
+					'AWS_S3_ENDPOINT' => "https://{$this->tld}/",
+				],
+				'external_links' => [
+					"proxy:s3-{$this->hostname}",
+				],
+			],
+		];
+	}
+	protected function get_service_mailhog() : array {
+		return [
+			'mailhog' => [
+				'image' => 'mailhog/mailhog:latest',
+				'ports' => [
+					'8025',
+					'1025',
+				],
+				'networks' => [
+					'proxy',
+					'default',
+				],
+				'labels' => [
+					'traefik.port=8025',
+					'traefik.protocol=http',
+					'traefik.docker.network=proxy',
+					"traefik.frontend.rule=Host:{$this->hostname};PathPrefix:/mailhog",
+				],
+				'environment' => [
+					'MH_UI_WEB_PATH' => 'mailhog',
+				],
+			],
+		];
+	}
+	protected function get_service_analytics() : array {
+		return [
+			'cognito' => [
+				'ports' => [
+					'3000',
+				],
+				'networks' => [
+					'proxy',
+					'default',
+				],
+				'restart' => 'on-failure',
+				'image' => 'humanmade/local-cognito:1.0.0',
+				'labels' => [
+					'traefik.port=3000',
+					'traefik.protocol=http',
+					'traefik.docker.network=proxy',
+					"traefik.frontend.rule=Host:cognito-{$this->hostname}",
+				],
+			],
+			'pinpoint' => [
+				'ports' => [
+					'3000',
+				],
+				'networks' => [
+					'proxy',
+					'default',
+				],
+				'restart' => 'on-failure',
+				'image' => 'humanmade/local-pinpoint:1.2.2',
+				'labels' => [
+					'traefik.port=3000',
+					'traefik.protocol=http',
+					'traefik.docker.network=proxy',
+					"traefik.frontend.rule=Host:pinpoint-{$this->hostname}",
+				],
+				'environment' => [
+					'INDEX_ROTATION' => 'OneDay',
+				],
+			],
+		];
+	}
+	protected function get_service_xray() : array {
+		return [
+			'xray' => [
+				'image' => 'amazon/aws-xray-daemon:3.0.1',
+				'ports' => [
+					'2000',
+				],
+				'environment' => [
+					'AWS_ACCESS_KEY_ID' => 'YOUR_KEY_HERE',
+					'AWS_SECRET_ACCESS_KEY' => 'YOUR_SECRET_HERE',
+					'AWS_REGION' => 'us-east-1',
+				],
+			],
+		];
+	}
+	public function get_array() : array {
+		$services = array_merge(
+			$this->get_service_db(),
+			$this->get_service_redis(),
+			$this->get_service_php(),
+			$this->get_service_nginx()
+		);
+
+		if ( $this->get_config()['xray'] ) {
+			$services = array_merge( $services, $this->get_service_xray() );
+		}
+
+		if ( $this->get_config()['cavalcade'] ) {
+			$services = array_merge( $services, $this->get_service_cavalcade() );
+		}
+
+		if ( $this->get_config()['elasticsearch'] ) {
+			$services = array_merge( $services, $this->get_service_elasticsearch() );
+		}
+
+		$services = array_merge(
+			$services,
+			$this->get_service_s3(),
+			$this->get_service_tachyon(),
+			$this->get_service_mailhog(),
+		);
+		if ( $this->get_config()['analytics'] ) {
```

Comment:
Analytics depends on elasticsearch so that should be factored in here
```suggestion
		if ( $this->get_config()['analytics'] && $this->get_config()['elasticsearch'] ) {
```